docs/website/contents/for-developers/HardForkCombinatorRollbackMonotonicityProposal.md

```diff
@@ -0,0 +1,285 @@
+        GetPoolState                            :: Maybe (Set (SL.KeyHash 'SL.StakePool (EraCrypto era))) -> BlockQuery (ShelleyBlock proto era) (SL.PState era)
+        GetStakeSnapshots                       :: Maybe (Set (SL.KeyHash 'SL.StakePool (EraCrypto era))) -> BlockQuery (ShelleyBlock proto era) (StakeSnapshots (EraCrypto era))
+        GetPoolDistr                            :: Maybe (Set (SL.KeyHash 'SL.StakePool (EraCrypto era))) -> BlockQuery (ShelleyBlock proto era) (SL.PoolDistr (EraCrypto era))
+        GetStakeDelegDeposits                   :: Set (StakeCredential (EraCrypto era)) -> BlockQuery (ShelleyBlock proto era) (Map (StakeCredential (EraCrypto era)) Coin)
+        GetConstitution                         :: BlockQuery (ShelleyBlock proto era) (Maybe (LC.Constitution era))
+        GetGovState                             :: BlockQuery (ShelleyBlock proto era) (LC.GovState era)
+        GetDRepState                            :: Set (SL.Credential 'DRepRole (EraCrypto era)) -> BlockQuery (ShelleyBlock proto era) (Map (SL.Credential 'DRepRole (EraCrypto era)) (SL.DRepState (EraCrypto era)))
+        GetDRepStakeDistr                       :: Set (LC.DRep (EraCrypto era)) -> BlockQuery (ShelleyBlock proto era) (Map (LC.DRep (EraCrypto era)) Coin)
+        GetCommitteeState                       :: BlockQuery (ShelleyBlock proto era) (SL.CommitteeState era)
+
+The queries marked with `*` and/or `+` are the only queries whose responses exhibit weak monotinicity as the node improves its selection by switching/extending chains.
+For example, once `GetChainBlockNo` becomes `NotOrigin x`, it never afterwards return `NotOrigin y` where `y < x`.
+Similarly, once `GetEraStart` for some era of interest becomes `Just x`, it will remain `Just x`.
+
+- `++` means the monotonicity is absolute (eg it's a constant value for this blockchain).
+- `**` means the monotonicity could only be spoiled if something like data corruption caused the node to roll back some blocks without simultaneously adding at least that many blocks.
+- `*+` means the monotonicity could only be spoiled if the node somehow performed a switch that rolled back more than k blocks.
+   Call this _k-rollback-monotonicity_.
+
+If the hard fork combinator were to always agree with the ledger---thereby avoiding the awkward surprise arising possible after a Chain Growth violation---then it would have to count only slots as the ledger does and not blocks.
+As a result, `GetEraStart` and `GetInterpreter` would no longer exhibit weak monotonicity.
+Specifically, a particular Chain Growth violation could cause `GetEraStart` to yield `Just x` in one moment and then `Nothing` or `Just y` where `x /= y` at a later moment, despite the node never rolling back more than k blocks.
+(`GetInterpreter` would be more intricate to describe, but the nub is exactly the same.)
+
+It is remarkable that every query without a marker is not weakly monotonic, and no one expects them to be!
```

Comment:
I agree, but part of that is that "the UTxO is monotonic" seems incoherent even (what would be the (partial) order) :+1: